{
  "origin": "codeshovel",
  "repositoryName": "hadoop",
  "repositoryPath": "/home/shaiful/research/codeshovel/codeshovel-projects/hadoop/.git",
  "startCommitName": "HEAD",
  "sourceFileName": "RPCUtil.java",
  "functionName": "instantiateIOException",
  "functionId": "instantiateIOException___cls-Class__? extends T____re-RemoteException",
  "sourceFilePath": "hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/ipc/RPCUtil.java",
  "functionStartLine": 78,
  "functionEndLine": 81,
  "numCommitsSeen": 10,
  "timeTaken": 1407,
  "changeHistory": [
    "87130bf6b22f538c5c26ad5cef984558a8117798"
  ],
  "changeHistoryShort": {
    "87130bf6b22f538c5c26ad5cef984558a8117798": "Yintroduced"
  },
  "changeHistoryDetails": {
    "87130bf6b22f538c5c26ad5cef984558a8117798": {
      "type": "Yintroduced",
      "commitMessage": "YARN-3400. [JDK 8] Build Failure due to unreported exceptions in RPCUtil (rkanter)\n",
      "commitDate": "26/03/15 11:00 AM",
      "commitName": "87130bf6b22f538c5c26ad5cef984558a8117798",
      "commitAuthor": "Robert Kanter",
      "diff": "@@ -0,0 +1,4 @@\n+  private static \u003cT extends IOException\u003e T instantiateIOException(\n+      Class\u003c? extends T\u003e cls, RemoteException re) throws RemoteException {\n+    return instantiateException(cls, re);\n+  }\n\\ No newline at end of file\n",
      "actualSource": "  private static \u003cT extends IOException\u003e T instantiateIOException(\n      Class\u003c? extends T\u003e cls, RemoteException re) throws RemoteException {\n    return instantiateException(cls, re);\n  }",
      "path": "hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/ipc/RPCUtil.java"
    }
  }
}